src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -188,6 +189,26 @@ public DateOnly AddDays(int value)
         /// <returns>true if left is the same as or earlier than right; otherwise, false.</returns>
         public static bool operator <=(DateOnly left, DateOnly right) => left._dayNumber <= right._dayNumber;
 
+        /// <summary>
+        /// Deconstructs <see cref="DateOnly"/> by <see cref="Year"/>, <see cref="Month"/> and <see cref="Day"/>.
+        /// </summary>
+        /// <param name="year">
+        /// Deconstructed parameter for <see cref="Year"/>.
+        /// </param>
+        /// <param name="month">
+        /// Deconstructed parameter for <see cref="Month"/>.
+        /// </param>
+        /// <param name="day">
+        /// Deconstructed parameter for <see cref="Day"/>.
+        /// </param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Deconstruct(out int year, out int month, out int day)
+        {
+            year = Year;
+            month = Month;
+            day = Day;
```

Comment:
The whole point of the destructor is to optimize the calculation. So what you are doing here is repeating the calculations 3 times. what you need to do here is just calling [https://github.com/dotnet/runtime/blob/7e10f81f6cd52ed14208373ac63b9ebd8296a62d/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L1351](https://github.com/dotnet/runtime/blob/7e10f81f6cd52ed14208373ac63b9ebd8296a62d/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L1351)

```C#
GetEquivalentDateTime().GetDate(out year, out month, out day);
```

I'll wait your change and then will continue the review after that. Thanks!